api/routes/auth.js

```diff
@@ -0,0 +1,107 @@
+const router = require("express").Router();
+const Users = require("../models/user");
+const bcrypt = require("bcrypt");
+const { sign, verify } = require("jsonwebtoken");
+
+const { SECRET_KEY } = process.env;
+
+//Creates an account - validate username and password, return JWT
+router.post("/signup", async (req, res, next) => {
+  const status = 201;
+
+  try {
+    const { username, password } = req.body;
+    //Check for existing user
+    let user = await Users.findOne({ username });
+    if (user) throw new Error(`User ${username} already exists`);
+    //Check for password min length
+    if (req.body.password.length < 8)
+      throw new Error("Password must be atleast 8 characters long!");
+    const saltRounds = 10;
+    const hashedPassword = await bcrypt.hash(password, saltRounds);
+    user = await Users.create({
+      username,
+      password: hashedPassword,
+      admin: req.body.admin
```

Comment:
Technically, this allows someone to make a request to sign up and then set themselves as an admin. For a real application, you may want to always set the default to `false` and then only allow for people to be updated to be an admin.